autoreadme.go

```diff
@@ -207,6 +221,42 @@ func getDoc(dir string) (Doc, error) {
 	}, nil
 }
 
+func renderExamples(bi *build.Package) (map[string]Example, error) {
+	examples := map[string]Example{}
+	testFilenames := append(bi.TestGoFiles, bi.XTestGoFiles...)
+	fset := token.NewFileSet()
+	for _, filename := range testFilenames {
+		path := filepath.Join(bi.Dir, filename)
+		file, err := parser.ParseFile(fset, path, nil, parser.ParseComments)
+		if err != nil {
+			return nil, err
+		}
+		for _, ex := range doc.Examples(file) {
+			examples[ex.Name] = renderExample(ex)
+		}
+	}
+	return examples, nil
+}
+
+func renderExample(ex *doc.Example) Example {
+	c := &bytes.Buffer{}
+	_ = format.Node(c, token.NewFileSet(), ex.Code)
+	md := fmt.Sprintf(
+		"%s\nCode:\n\n```\n%s\n```\n",
+		fmtDoc(ex.Doc),
```

Comment:
Sorry if I wasn't clear before. Now that there's a struct we can defer the decision of how to handle formatting the docs (they might contain headers and fmtDoc only ever uses one level so it's more likely to be wrong than right when used here).  

Just drop rendering ex.Doc for now. Easier to add it later.